src/Common/Core/Header/Cookie.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Common\Core\Header;
+
+use Common\Core\Model;
+use Symfony\Component\HttpFoundation\Cookie as SymfonyCookie;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+final class Cookie
+{
+    /**
+     * @var ResponseHeaderBag
+     */
+    private $newCookiesHeaderBag;
+
+    public function __construct()
+    {
+        $this->newCookiesHeaderBag = new ResponseHeaderBag();
+    }
+
+    /**
+     * Creates a new instance of the Symfony cookie
+     *
+     * It will automatically make them secure if you are on https and will set the domain if it isn't set
+     *
+     * @param string $name The name of the cookie
+     * @param string|null $value The value of the cookie
+     * @param int $expire The number of seconds before the cookie expires, defaults to 30 days
+     * @param string $path The path on the server in which the cookie will be available on
+     * @param string|null $domain The domain that the cookie is available to
+     * @param bool|null $secure Whether the cookie should only be transmitted over a secure HTTPS connection from the client. When null is past it will be secure only when set on https
+     * @param bool $httpOnly Whether the cookie will be made accessible only through the HTTP protocol
```

Comment:
because sometimes you do need to set a cookie when on https that also works on http and this makes that possible